source/patterns/@aws-solutions-constructs/core/lib/kendra-helper.ts

```diff
@@ -0,0 +1,254 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+/*
+ *  The functions found here in the core library are for internal use and can be changed
+ *  or removed outside of a major release. We recommend against calling them directly from client code.
+ */
+
+import * as kendra from 'aws-cdk-lib/aws-kendra';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import { addCfnSuppressRules, consolidateProps } from "./utils";
+import { Aws } from 'aws-cdk-lib';
+
+// Note: To ensure CDKv2 compatibility, keep the import statement for Construct separate
+import { Construct } from 'constructs';
+import { DefaultKendraIndexProps } from './kendra-defaults';
+import { generatePhysicalName, overrideProps } from "./utils";
+
+export interface BuildKendraIndexProps {
+  readonly kendraIndexProps?: kendra.CfnIndexProps | any;
+  /**
+   * Existing instance of Kendra Index object, Providing both this and kendraIndexProps will cause an error.
+   *
+   * @default - None
+   */
+  readonly existingIndexObj?: kendra.CfnIndex;
+}
+
+/**
+ * @internal This is an internal core function and should not be called directly by Solutions Constructs clients.
+ */
+export function buildKendraIndex(scope: Construct, id: string, props: BuildKendraIndexProps): kendra.CfnIndex {
+  // Conditional lambda function creation
+  if (props.existingIndexObj) {
+    // The client provided an Index, so we'll do nothing and return it to them
+    return props.existingIndexObj;
```

Comment:
Where do we do the error checking that specifying both `kendraIndexProps` and `existingIndexObj` throws an error? I may find it as I continue reviewing.